refactor: update `blas/base/gscal` to follow current project conventions

saurabhraghuvanshii opened the pull request:
Resolves #5440.

## Description

> What is the purpose of this pull request?

This pull request:

-  refactor the package to follow current project conventions

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5440 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

saurabhraghuvanshii committed:
refactor: update to follow current project convetions

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/README.md:
```diff
@@ -20,7 +20,7 @@ limitations under the License.
 # gscal
 
-> Multiply a vector `x` by a constant `alpha`.
+> Multiply a vector `x` by a scalar constant `alpha`.
```
```suggestion
> Multiply a vector by a scalar constant.
```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/README.md:
````diff
@@ -30,9 +30,9 @@ limitations under the License.
 ```
 
-#### gscal( N, alpha, x, stride )
+#### gscal( N, alpha, x, strideX )
````
```suggestion
#### gscal( N, alpha, x, stride )
```
When there's only one input, it's okay to stride instead of strideX

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/README.md:
```diff
@@ -30,9 +30,9 @@ limitations under the License.
+#### gscal( N, alpha, x, strideX )
 
-Multiplies a vector `x` by a constant `alpha`.
+Multiplies a vector `x` by a scalar constant `alpha`.
```
```suggestion
Multiplies a vector by a scalar constant.
```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/README.md:
```diff
@@ -46,9 +46,9 @@ The function has the following parameters:
 -   **alpha**: scalar constant.
 -   **x**: input [`Array`][mdn-array] or [`typed array`][mdn-typed-array].
--   **stride**: index increment.
+-   **strideX**: stride length.
```
```suggestion
-   **stride**: stride length.
```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/README.md:
```diff
@@ -46,9 +46,9 @@ The function has the following parameters:
 
-The `N` and `stride` parameters determine which elements in `x` are accessed at runtime. For example, to multiply every other value by a constant
+The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to multiply every other value by a scalar constant
 
```
```suggestion
The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to multiply every other value by a scalar constant:

```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/README.md:
```diff
@@ -75,9 +75,9 @@ gscal( 3, 5.0, x1, 2 );
 If either `N` or `stride` is less than or equal to `0`, the function returns `x` unchanged.
 
-#### gscal.ndarray( N, alpha, x, stride, offset )
+#### gscal.ndarray( N, alpha, x, strideX, offsetX )
```
```suggestion
#### gscal.ndarray( N, alpha, x, stride, offset )
```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/README.md:
```diff
@@ -75,9 +75,9 @@ gscal( 3, 5.0, x1, 2 );
 
-Multiplies a vector `x` by a constant `alpha` using alternative indexing semantics.
+Multiplies a vector `x` by a scalar constant `alpha` using alternative indexing semantics.
 
```
```suggestion
Multiplies a vector by a scalar constant using alternative indexing semantics.

```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/README.md:
```diff
@@ -88,9 +88,9 @@ gscal.ndarray( x.length, 5.0, x, 1, 0 );
 The function has the following additional parameters:
 
--   **offset**: starting index.
+-   **offsetX**: starting index.
```
```suggestion
-   **offset**: starting index.
```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/lib/ndarray.js:
```diff
@@ -37,8 +37,8 @@ var M = 5;
-* @param {integer} stride - index increment
-* @param {NonNegativeInteger} offset - starting index
+* @param {integer} strideX - stride length
+* @param {NonNegativeInteger} offsetX - starting index
```
```suggestion
* @param {integer} stride - stride length
* @param {NonNegativeInteger} offset - starting index
```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/lib/ndarray.js:
```diff
@@ -47,7 +47,7 @@ var M = 5;
 * // x => [ 1.0, -2.0, 3.0, -20.0, 25.0, -30.0 ]
 */
-function gscal( N, alpha, x, stride, offset ) {
+function gscal( N, alpha, x, strideX, offsetX ) {
```
Same suggestion throughout this file.

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/lib/main.js:
```diff
@@ -30,9 +30,9 @@ var ndarray = require( './ndarray.js' );
+* @param {number} alpha - scalar constant
 * @param {NumericArray} x - input array
-* @param {PositiveInteger} stride - index increment
+* @param {PositiveInteger} strideX -  stride length
```
```suggestion
* @param {PositiveInteger} stride -  stride length
```
Same suggestion throughout this file.

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/lib/main.js:
```diff
@@ -30,9 +30,9 @@ var ndarray = require( './ndarray.js' );
 * Multiplies `x` by a scalar `alpha`.
```
```suggestion
* Multiplies a vector by a scalar constant.
```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/lib/ndarray.js:
```diff
@@ -37,8 +37,8 @@ var M = 5;
 * @param {PositiveInteger} N - number of indexed elements
```
Need to change the function description here.

saurabhraghuvanshii replied:
sorry , I didn't get it .

headlessNode replied:
The function description on Line35.

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/lib/index.js:
```diff
@@ -19,7 +19,7 @@
 
 /**
-* BLAS level 1 routine to multiply `x` by a constant.
+* BLAS level 1 routine to multiply `x` by a scalar constant.
```
```suggestion
* Multiply a vector by a scalar constant.
```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/docs/repl.txt:
```diff
@@ -1,28 +1,28 @@
 
-{{alias}}( N, alpha, x, stride )
-    Multiplies a vector `x` by a constant `alpha`.
+{{alias}}( N, alpha, x, strideX )
```
```suggestion
{{alias}}( N, alpha, x, stride )
```
Same suggestion throughout this file.

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/docs/repl.txt:
```diff
@@ -1,28 +1,28 @@
-{{alias}}( N, alpha, x, stride )
-    Multiplies a vector `x` by a constant `alpha`.
+{{alias}}( N, alpha, x, strideX )
+    Multiplies a vector `x` by a scalar constant `alpha`.
```
```suggestion
    Multiplies a vector by a scalar constant
```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/docs/repl.txt:
```diff
@@ -51,11 +50,11 @@
-    buffer, the `offset` parameter supports indexing semantics based on a
+    buffer, the offset parameter supports indexing semantics based on a
     starting index.
 
```
```suggestion
    buffer, the offset parameter supports indexing semantics based on a starting
    index.

```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/docs/types/index.d.ts:
```diff
@@ -32,12 +32,12 @@ type InputArray = NumericArray | Collection<number> | AccessorArrayLike<number>;
 interface Routine {
 	/**
-	* Multiplies a vector `x` by a constant `alpha`.
+	* Multiplies a vector `x` by a scalar constant `alpha`.
```
```suggestion
	* Multiplies a vector by a scalar constant.
```
Same suggestion throughout this file.

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/docs/types/index.d.ts:
```diff
@@ -32,12 +32,12 @@ type InputArray = NumericArray | Collection<number> | AccessorArrayLike<number>;
+	* @param alpha - scalar constant
 	* @param x - input array
-	* @param stride - stride length
+	* @param strideX - stride length
```
```suggestion
	* @param stride - stride length
```
Same suggestion throughout this file.

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/docs/repl.txt:
```diff
@@ -51,11 +50,11 @@
-{{alias}}.ndarray( N, alpha, x, stride, offset )
-    Multiplies `x` by a constant `alpha` using alternative indexing semantics.
+{{alias}}.ndarray( N, alpha, x, strideX, offsetX )
+    Multiplies `x` by a scalar constant `alpha` using alternative indexing semantics.
```
This line is more than `80` characters.

headlessNode reviewed (changes requested):
Thanks for working on this @saurabhraghuvanshii .

saurabhraghuvanshii committed:
chore: address changes

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

saurabhraghuvanshii committed:
Merge branch 'stdlib-js:develop' into refactor-gscal

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/README.md:
```diff
@@ -32,7 +32,7 @@ var gscal = require( '@stdlib/blas/base/gscal' );
 #### gscal( N, alpha, x, stride )
 
-Multiplies a vector `x` by a constant `alpha`.
+Multiply a vector by a scalar constant.
```
```suggestion
Multiplies a vector by a scalar constant.
```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/docs/repl.txt:
```diff
@@ -1,9 +1,9 @@
 
 {{alias}}( N, alpha, x, stride )
-    Multiplies a vector `x` by a constant `alpha`.
+    Multiplies a vector by a scalar constant
```
```suggestion
    Multiplies a vector by a scalar constant.
```

saurabhraghuvanshii committed:
chore: addresses comments

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

saurabhraghuvanshii committed:
Merge branch 'stdlib-js:develop' into refactor-gscal

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/docs/types/index.d.ts:
```diff
@@ -49,13 +49,13 @@ interface Routine {
 	* @param x - input array
 	* @param stride - stride length
-	* @param offset - starting index
+	* @param offsetX - starting index
```
```suggestion
	* @param offset - starting index
```

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/docs/types/index.d.ts:
```diff
@@ -64,14 +64,14 @@ interface Routine {
 	* // x => [ -10.0, 5.0, 15.0, -25.0, 20.0, 0.0, -5.0, -15.0 ]
 	*/
-	ndarray<T extends InputArray>( N: number, alpha: number, x: T, stride: number, offset: number ): T;
+	ndarray<T extends InputArray>( N: number, alpha: number, x: T, stride: number, offsetX: number ): T;
```
```suggestion
	ndarray<T extends InputArray>( N: number, alpha: number, x: T, stride: number, offset: number ): T;
```

headlessNode committed:
docs: apply suggestions from code review

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
refactor: apply suggestions from code review

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode commented on lib/node_modules/@stdlib/blas/base/gscal/docs/repl.txt:
```diff
@@ -32,13 +32,12 @@
-    // Using `N` and `stride` parameters:
-    > x = [ -2.0, 1.0, 3.0, -5.0, 4.0, -1.0, -3.0 ];
+    // Using `N` and stride parameters:
+    > x = [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0 -1.0, -3.0 ];
```
```suggestion
    > x = [ -2.0, 1.0, 3.0, -5.0, 4.0, -1.0, -3.0 ];
```

headlessNode committed:
docs: apply suggestions from code review

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into refactor-gscal

kgryte commented on lib/node_modules/@stdlib/blas/base/gscal/docs/repl.txt:
```diff
@@ -51,29 +50,29 @@
-{{alias}}.ndarray( N, alpha, x, stride, offset )
-    Multiplies `x` by a constant `alpha` using alternative indexing semantics.
+{{alias}}.ndarray( N, alpha, x, stride, offsetX )
+    Multiplies a vector by a scalar constant.
```
```suggestion
    Multiplies a vector by a scalar constant using alternative indexing semantics.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/gscal/docs/repl.txt:
```diff
@@ -51,29 +50,29 @@
-{{alias}}.ndarray( N, alpha, x, stride, offset )
-    Multiplies `x` by a constant `alpha` using alternative indexing semantics.
+{{alias}}.ndarray( N, alpha, x, stride, offsetX )
+    Multiplies a vector by a scalar constant using alternative indexing semantics.
```
```suggestion
    Multiplies a vector by a scalar constant using alternative indexing
    semantics.
```

kgryte committed:
style: wrap text

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: clean-up

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/gscal/docs/types/index.d.ts:
```diff
@@ -49,10 +49,10 @@ interface Routine {
 
 	/**
-	* Multiplies a vector `x` by a constant `alpha` using alternative indexing semantics.
+	* Multiplies a vector by a scalar constant.
```
```suggestion
	* Multiplies a vector by a scalar constant using alternative indexing semantics.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/gscal/lib/ndarray.js:
```diff
@@ -32,12 +32,12 @@ var M = 5;
 
 /**
-* Multiplies `x` by a scalar `alpha`.
+* Multiplies a vector by a scalar constant.
```
```suggestion
* Multiplies a vector by a scalar constant using alternative indexing semantics.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>
